### DISCUSSION TITLE
Welcome to box-python-sdk Discussions!

### DISCUSSION THREAD

**Discussion by mgrytsai** [box/box-python-sdk#731] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by annie0467** [comment]
Given a single folder, Is there any way to keep monitoring all of its subfolders for events(ex: ITEM_UPLOAD, ITEM_DOWNLOAD, etc) and associate it with the specific folder name?

For ex: hierarchy is as follows
/ Main_Folder
-> A
-> B
-> C

There might be more folders in A, B, C
I need to track events in each subfolder folder

I wish to use generate_events_with_long_polling()

I have tried creating a monitoring thread on each single folder, but I run into unexpected issues after sometime.

**Reply by mgrytsai** [reply]
I see that you created an issue for the same. Let me include a [link here](https://github.com/box/box-python-sdk/issues/828#issuecomment-1632451452), so anyone in the future could find it.
